src/super_gradients/training/datasets/detection_datasets/coco_detection_yolox.py

```diff
@@ -0,0 +1,287 @@
+        self.annotations = self._load_coco_annotations()
+        if cache:  # cache after merged
+            self._cache_images()
+
+        self.transforms = transforms
+
+    def __getitem__(self, idx):
+        sample = self.load_sample(idx)
+        sample = self.apply_transforms(sample)
+        return sample["image"], sample["target"], sample["info"], sample["id"]
+
+    def __len__(self):
+        return len(self.ids)
+
+    def _load_coco_annotations(self):
+        return [self._load_anno_from_ids(_ids) for _ids in self.ids]
+
+    def _load_anno_from_ids(self, id_):
+        im_ann = self.coco.loadImgs(id_)[0]
+        width = im_ann["width"]
+        height = im_ann["height"]
+        anno_ids = self.coco.getAnnIds(imgIds=[int(id_)], iscrowd=False)
+        annotations = self.coco.loadAnns(anno_ids)
+        objs = []
+        for obj in annotations:
+            x1 = np.max((0, obj["bbox"][0]))
+            y1 = np.max((0, obj["bbox"][1]))
+            x2 = np.min((width, x1 + np.max((0, obj["bbox"][2]))))
+            y2 = np.min((height, y1 + np.max((0, obj["bbox"][3]))))
+            if obj["area"] > 0 and x2 >= x1 and y2 >= y1:
+                obj["clean_bbox"] = [x1, y1, x2, y2]
+                objs.append(obj)
+
+        num_objs = len(objs)
+
+        res = np.zeros((num_objs, 5))
+        num_seg_values = 98 if self.tight_box_rotation else 0
+        res_seg = np.ones((num_objs, num_seg_values))
+        res_seg.fill(np.nan)
+
+        for ix, obj in enumerate(objs):
+            cls = self.class_ids.index(obj["category_id"])
+            res[ix, 0:4] = obj["clean_bbox"]
+            res[ix, 4] = cls
+            if self.tight_box_rotation:
+                seg_points = [j for i in obj.get("segmentation", []) for j in i]
+                if seg_points:
+                    seg_points_c = np.array(seg_points).reshape((-1, 2)).astype(np.int)
+                    seg_points_convex = cv2.convexHull(seg_points_c).ravel()
+                else:
+                    seg_points_convex = []
+                res_seg[ix, :len(seg_points_convex)] = seg_points_convex
+
+        r = min(self.input_dim[0] / height, self.input_dim[1] / width)
+        res[:, :4] *= r
+        res_seg *= r
+
+        img_info = (height, width)
+        resized_info = (int(height * r), int(width * r))
+
+        file_name = (
+            im_ann["file_name"]
+            if "file_name" in im_ann
+            else "{:012}".format(id_) + ".jpg"
+        )
+
+        return res, res_seg, img_info, resized_info, os.path.join(self.data_dir, self.name, file_name)
+
+    def _cache_images(self):
+        logger.warning(
+            "\n********************************************************************************\n"
+            "You are using cached images in RAM to accelerate training.\n"
+            "This requires large system RAM.\n"
+            "Make sure you have 200G+ RAM and 136G available disk space for training COCO.\n"
```

Comment:
an important thing to note. if you are using DDP, this will be 8 times more memory!

Reply:
Not true, once the cache file is written (which is done once)- the different processes just sample from it according to the sampler.